stripe/src/test/java/com/stripe/android/StripeKotlinTest.kt

```diff
@@ -0,0 +1,678 @@
+package com.stripe.android
+
+import android.content.Intent
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.stripe.android.exception.APIException
+import com.stripe.android.exception.AuthenticationException
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.Source
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeApiRepository
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.mockito.Mockito.`when`
+import kotlin.test.assertFailsWith
+import kotlin.test.assertSame
+
+/**
+ * Test for [Stripe] suspend functions.
+ */
+@RunWith(JUnit4::class)
+internal class StripeKotlinTest {
+    private val mockApiRepository: StripeApiRepository = mock()
+    private val mockPaymentController: PaymentController = mock()
+
+    private val stripe: Stripe =
+        Stripe(mockApiRepository, mockPaymentController, TEST_PUBLISHABLE_KEY)
+
+    @Test
+    fun `When repository returns correct value then createPaymentMethod should Succeed`(): Unit =
+        `Given repository returns non-empty value when calling createAPI then returns correct result`(
+            mockApiRepository::createPaymentMethod,
+            stripe::createPaymentMethod
+        )
```

Comment:
I think this pattern is a bit hard to follow.

Reply:
could you elaborate? do you mean the function name pattern?

Reply:
Yeah, can we use "standard" function names if we're directly calling?

Reply:
Resolved after offline sync